src/ILCompiler.Compiler/src/Compiler/NameMangler.cs

```diff
@@ -345,5 +348,40 @@ private string ComputeMangledFieldName(FieldDesc field)
 
             return mangledName;
         }
+
+        private ImmutableDictionary<string, string> _mangledStringLiterals = ImmutableDictionary<string, string>.Empty;
+
+        public string GetMangledStringName(string literal)
```

Comment:
The NameMangler also ensures the names we pick for symbols / C++ identifiers are valid by sanitizing them which is what we are doing with string literal names, so I would say it does make sense to put this here.